aaronmassey/emul sbs gauge reg rw improvements

aaronemassey opened the pull request:
Emulator improvements that makeup the first two commits of https://github.com/zephyrproject-rtos/zephyr/pull/46817

Once this merges, I'll rebase that PR's branch on top of main so make it a smaller PR.

aaronemassey committed:
emul: emul_sbs_gauge read/write returns ret code

To facilitate testing improper reads and writes on various registers,
enable the emul_sbs_gauge reg_read/write handlers to return custom exit
codes.

Signed-off-by: Aaron Massey <aaronmassey@google.com>

aaronemassey committed:
emul_sbs_gauge: Rename reg_read/write more unique

Rename the reg_read and reg_write functions in the emul_sbs_gauge emulator
to be more unique symbols so as to support overriding this specific
function in future tests that make use of FFF. See #46817 which includes a
test for error reads as an example.

Signed-off-by: Aaron Massey <aaronmassey@google.com>

yperess commented on subsys/emul/i2c/emul_sbs_gauge.c:
```diff
@@ -102,7 +104,7 @@ static int sbs_gauge_emul_transfer_i2c(const struct emul *target, struct i2c_msg
 			case 1:
-				val = reg_read(target, reg);
+				rc = emul_sbs_gauge_reg_read(target, reg, &val);
 				msgs->buf[0] = val;
```
I think this needs to check `rc` first

fabiobaltieri commented on subsys/emul/i2c/emul_sbs_gauge.c:
```diff
@@ -61,15 +62,15 @@ static int reg_read(const struct emul *target, int reg)
 		LOG_ERR("Unknown register 0x%x read", reg);
 		return -EIO;
 	}
 	LOG_INF("read 0x%x = 0x%x", reg, val);
```
```
/__w/zephyr/zephyr/subsys/emul/i2c/emul_sbs_gauge.c: In function 'emul_sbs_gauge_reg_read':
/__w/zephyr/zephyr/subsys/emul/i2c/emul_sbs_gauge.c:71:17: error: format '%x' expects argument of type 'unsigned int', but argument 3 has type 'int *' [-Werror=format=]
   71 |         LOG_INF("read 0x%x = 0x%x", reg, val);
      |                 ^~~~~~~~~~~~~~~~~~       ~~~
      |                                          |
      |                                          int *
```

aaronemassey replied:
Fixed, but for some reason I didn't get this build failure locally. Do you have yours setup with `clang-tidy`?

fabiobaltieri replied:
Yeah I saw it on the CI run, one step was failing.

aaronemassey committed:
emul_sbs_gauge: Don't set bad val in i2c msg buf

The SBS Gauge emulator would set the I2C buffer message with a value from
the emulator's custom read function even on a read error instead of just
returning. This behavior may be improperly relied upon by future tests.

Return early on register read before setting the i2c message buffer value
in the emulator.

Signed-off-by: Aaron Massey <aaronmassey@google.com>

alevkoy reviewed (dismissed):
The function renames look fine to me, but could they actually be overridden by other symbols before? They appear to be strongly and statically linked.

aaronemassey commented:
> The function renames look fine to me, but could they actually be overridden by other symbols before? They appear to be strongly and statically linked.

I'll clarify in the commit message that this is supposed to support changes like https://github.com/zephyrproject-rtos/zephyr/pull/46817/commits/03d1ff64bc92aa1fa6cc70fed53869009062508a where it may be changed to test_mockable.

aaronemassey commented:
Adding DNM as there's a test I'd like to do locally real quick that looked like I was skipping. Removing momentarily.

aaronemassey commented:
Removed DNM, all is well. This is ready to merge.
